File: PasswordGenerator.py
import string
import random

# Function to get user input and validate it is a non-negative number
def get_user_input(prompt):
    while True:
        try:
            user_input = int(input(prompt))
            if user_input < 0:
                print("Your number should be non-negative.")
            else:
                return user_input
        except ValueError:
            print("Please enter numbers only.")

print("Welcome to Password Generator")

# Ask user about the number of characters
num_letters = get_user_input("How many letters do you want in your password? ")
num_numbers = get_user_input("How many numbers do you want in your password? ")
num_symbols = get_user_input("How many symbols do you want in your password? ")

# Using some predefined functions from the respective library
letters_upper = string.ascii_uppercase
letters_lower = string.ascii_lowercase
numbers = string.digits
symbols = string.punctuation

# Using random.choices() to allow repetition
letters = random.choices(letters_upper + letters_lower, k=num_letters)
nums = random.choices(numbers, k=num_numbers)
syms = random.choices(symbols, k=num_symbols)

# Concatenate the lists and shuffle them
password_list = letters + nums + syms
random.shuffle(password_list)

# Join the shuffled list elements to create the password
password = "".join(password_list)

# Output the generated password
print('This is your password requirements:', password)

File: PasswordGenerator.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <ctime>
#include <random>

using namespace std;

int main() {
    cout << "Welcome to Password Generator" << endl;

    int num_letters, num_numbers, num_symbols;
    cout << "How many letters do you want in your password? ";
    cin >> num_letters;

    cout << "How many numbers do you want in your password? ";
    cin >> num_numbers;

    cout << "How many symbols do you want in your password? ";
    cin >> num_symbols;

    string letters_upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    string letters_lower = "abcdefghijklmnopqrstuvwxyz";
    string numbers = "0123456789";
    string symbols = "!@#$%^&*()_+-=[]{}|;:',.<>?";

    // Seed the random number generator using time
    mt19937 generator(time(0));

    // Shuffle each category separately
    shuffle(letters_upper.begin(), letters_upper.end(), generator);
    shuffle(letters_lower.begin(), letters_lower.end(), generator);
    shuffle(numbers.begin(), numbers.end(), generator);
    shuffle(symbols.begin(), symbols.end(), generator);

    // Select the desired number of characters from each category
    string password = letters_upper.substr(0, num_letters) +
                      letters_lower.substr(0, num_letters) +
                      numbers.substr(0, num_numbers) +
                      symbols.substr(0, num_symbols);

    // Shuffle the password string to ensure randomness
    shuffle(password.begin(), password.end(), generator);

    // Output the generated password
    cout << "This is your password requirements:😎 " << password << endl;

    return 0;
}

